File: order-service/src/main/java/com/intexsoft/webshop/orderservice/model/Status.java
package com.intexsoft.webshop.orderservice.model;

import com.intexsoft.webshop.orderservice.model.enums.StatusName;
import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.SourceType;
import org.hibernate.proxy.HibernateProxy;

import java.time.LocalDateTime;
import java.util.Objects;

@Entity
@Table(name = "status")
@Getter
@Setter
@ToString
@Builder
@AllArgsConstructor
@NoArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class Status {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Long id;
    @CreationTimestamp(source = SourceType.DB)
    @Column(name = "created_at", nullable = false)
    LocalDateTime createdAt;
    @Column(name = "name", nullable = false)
    @Enumerated(EnumType.STRING)
    StatusName statusName;
    @ManyToOne(fetch = FetchType.LAZY)
    @ToString.Exclude
    Order order;

    public Status(StatusName statusName) {
        this.statusName = statusName;
    }

    @Override
    public final boolean equals(Object object) {
        if (this == object) return true;
        if (object == null) return false;
        Class<?> oEffectiveClass = object instanceof HibernateProxy ? ((HibernateProxy) object).getHibernateLazyInitializer().getPersistentClass() : object.getClass();
        Class<?> thisEffectiveClass = this instanceof HibernateProxy ? ((HibernateProxy) this).getHibernateLazyInitializer().getPersistentClass() : this.getClass();
        if (thisEffectiveClass != oEffectiveClass) return false;
        Status status = (Status) object;
        return getId() != null && Objects.equals(getId(), status.getId());
    }

    @Override
    public final int hashCode() {
        return this instanceof HibernateProxy ? ((HibernateProxy) this).getHibernateLazyInitializer().getPersistentClass().hashCode() : getClass().hashCode();
    }
}

File: order-service-kt/src/main/kotlin/com/intexsoft/webshop/orderservicekt/model/Status.kt
package com.intexsoft.webshop.orderservicekt.model

import com.intexsoft.webshop.orderservicekt.model.enums.StatusName
import jakarta.persistence.*
import org.hibernate.annotations.CreationTimestamp
import org.hibernate.annotations.SourceType
import java.time.LocalDateTime

@Entity
@Table(name = "status")
class Status(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long = 0,
    @CreationTimestamp(source = SourceType.DB)
    @Column(name = "created_at", nullable = false)
    var createdAt: LocalDateTime? = null,
    @Column(name = "name", nullable = false)
    @Enumerated(EnumType.STRING)
    var statusName: StatusName = StatusName.INITIALIZED,
    @ManyToOne(fetch = FetchType.LAZY)
    var order: Order? = null
)